tests_e2e/tests/ext_policy/ext_policy.py

```diff
@@ -0,0 +1,287 @@
+
+    def __init__(self, context: AgentVmTestContext):
+        super().__init__(context)
+        self._ssh_client: SshClient = self._context.create_ssh_client()
+
+    def _create_policy_file(self, policy):
+        """
+        Create policy json file and copy to /etc/waagent_policy.json on test machine.
+        """
+        with open("waagent_policy.json", mode='w') as policy_file:
+            json.dump(policy, policy_file, indent=4)
+            policy_file.flush()
+
+            remote_path = "/tmp/waagent_policy.json"
+            local_path = policy_file.name
+            self._ssh_client.copy_to_node(local_path=local_path, remote_path=remote_path)
+            policy_file_final_dest = "/etc/waagent_policy.json"
+            log.info("Copying policy file to test VM [%s]", self._context.vm.name)
+            self._ssh_client.run_command(f"mv {remote_path} {policy_file_final_dest}", use_sudo=True)
+
+    def _operation_should_succeed(self, operation, extension_case):
+        log.info("")
+        log.info(f"Attempting to {operation} {extension_case.extension.__str__()}, expected to succeed ")
+        # Attempt operation. If enabling, assert that the extension is present in instance view.
+        # If deleting, assert that the extension is not present in instance view.
+        try:
+            if operation == "enable":
+                # VirtualMachineRunCommandClient (and VirtualMachineRunCommand) does not take force_update_tag as a parameter.
+                if isinstance(extension_case.extension, VirtualMachineRunCommandClient):
+                    extension_case.extension.enable(settings=extension_case.settings)
+                else:
+                    extension_case.extension.enable(settings=extension_case.settings, force_update=True)
+                extension_case.extension.assert_instance_view()
+            elif operation == "delete":
+                extension_case.extension.delete()
+                instance_view_extensions = self._context.vm.get_instance_view().extensions
+                if instance_view_extensions is not None and any(
+                        e.name == extension_case.extension._resource_name for e in instance_view_extensions):
+                    raise Exception(
+                        "extension {0} still in instance view after attempting to delete".format(extension_case.extension))
+            log.info(f"Operation '{operation}' for {extension_case.extension.__str__()} succeeded as expected.")
+        except Exception as error:
+            fail(
+                f"Unexpected error while trying to {operation} {extension_case.extension.__str__()}. "
+                f"Extension is allowed by policy so this operation should have completed successfully.\n"
+                f"Error: {error}")
+
+    def _operation_should_fail(self, operation, extension_case):
+        log.info("")
+        log.info(f"Attempting to {operation} {extension_case.extension}, should fail fast.")
```

Comment:
this should say "should reach timeout" for delete. split the log messages for enable and delete.

Reply:
done